vyxal/LazyList.py

```diff
@@ -128,19 +131,19 @@ def __setitem__(self, position, value):
         self.generated[position] = value
 
     def count(self, other):
+        """Number of occurrences of `other` in this `LazyList`"""
         temp = self.listify()
         return temp.count(other)
 
+    @lazylist
     def filter(self, fn):
-        @lazylist
-        def gen():
-            for item in self:
-                if fn(item):
-                    yield item
-
-        return gen()
+        """A `LazyList` containing only elements for whom `fn` is true"""
+        for item in self:
+            if fn(item):
+                yield item
 
     def listify(self):
+        """Turn into a list"""
```

Comment:
Use docstrings only where necessary to explain what something does. In this case it's pretty obvious.
```suggestion
```

Reply:
Deepsource was complaining about missing docstrings, we could add a comment to ignore that in the future